tasks/eth/006-MCP-L1/postCheck.s.sol

```diff
@@ -0,0 +1,231 @@
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+
+contract PostCheck is SignFromJson {
+    Types.ContractSet prox = Types.ContractSet({
+        L1CrossDomainMessenger: 0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1,
+        L1StandardBridge: 0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1,
+        L2OutputOracle: 0xdfe97868233d1aa22e815a266982f2cf17685a27,
+        DisputeGameFactory: address(0),
+        OptimismMintableERC20Factory: 0x75505a97BD334E7BD3C476893285569C4136Fa0F,
+        OptimismPortal: 0xbEb5Fc579115071764c7423A4f12eDde41f106Ed,
+        OptimismPortal2: 0xbEb5Fc579115071764c7423A4f12eDde41f106Ed,
+        SystemConfig: 0x229047fed2591dbec1eF1118d64F7aF3dB9EB290,
+        L1ERC721Bridge: 0x5a7749f83b81B301cAb5f48EB8516B986DAef23D,
+        ProtocolVersions: 0x8062AbC286f5e7D9428a0Ccb9AbD71e50d93b935,
+        SuperchainConfig: 0x95703e0982140D16f8ebA6d158FccEde42f04a4C
+    });
+    DeployConfig cfg = new DeployConfig();
+    uint256 l2OutputOracleStartingTimestamp = 1686068903;
+
+    constructor() {
+        // Load the DeployConfig from the JSON file
+        cfg.read(string.concat(vm.projectRoot(), "/tasks/eth/006-MCP-L1/DeployConfig.json"));
+    }
+
+    function _postCheck() internal view override {
+        postDeployAssertions();
+    }
+
+    /// @notice Asserts the correctness of an L1 deployment. This function expects that all contracts
+    ///         within the `prox` ContractSet are proxies that have been setup and initialized.
+    function postDeployAssertions() internal view {
+        console.log("Running post-deploy assertions");
+        ResourceMetering.ResourceConfig memory rcfg;
+        rcfg.maxResourceLimit = SystemConfig(prox.SystemConfig).resourceConfig().maxResourceLimit;
+        rcfg.elasticityMultiplier = SystemConfig(prox.SystemConfig).resourceConfig().elasticityMultiplier;
+        rcfg.baseFeeMaxChangeDenominator = SystemConfig(prox.SystemConfig).resourceConfig().baseFeeMaxChangeDenominator;
+        rcfg.minimumBaseFee = SystemConfig(prox.SystemConfig).resourceConfig().minimumBaseFee;
+        rcfg.systemTxMaxGas = SystemConfig(prox.SystemConfig).resourceConfig().systemTxMaxGas;
+        rcfg.maximumBaseFee = SystemConfig(prox.SystemConfig).resourceConfig().maximumBaseFee;
+        ResourceMetering.ResourceConfig memory dflt;
+        dflt.maxResourceLimit = Constants.DEFAULT_RESOURCE_CONFIG().maxResourceLimit;
+        dflt.elasticityMultiplier = Constants.DEFAULT_RESOURCE_CONFIG().elasticityMultiplier;
+        dflt.baseFeeMaxChangeDenominator = Constants.DEFAULT_RESOURCE_CONFIG().baseFeeMaxChangeDenominator;
```

Comment:
this could be modularized so not so many loc are needed? can just do 2 total calls? why split up